simtools/simtel/simtel_runner_ray_tracing.py

```diff
@@ -0,0 +1,256 @@
+        Contains the configurable parameters (zenithAngle).
+
+    Methods
+    -------
+    run(test=False, force=False)
+        Run sim_telarray. test=True will make it faster and force=True will remove existing files
+        and run again.
+    '''
+
+    def __init__(
+        self,
+        telescopeModel,
+        label=None,
+        simtelSourcePath=None,
+        filesLocation=None,
+        configData=None,
+        configFile=None,
+        singleMirrorMode=False
+    ):
+        '''
+        SimtelRunner.
+
+        Parameters
+        ----------
+        telescopeModel: str
+            Instance of TelescopeModel class.
+        label: str, optional
+            Instance label. Important for output file naming.
+        simtelSourcePath: str (or Path), optional
+            Location of sim_telarray installation. If not given, it will be taken from the
+            config.yml file.
+        filesLocation: str (or Path), optional
+            Parent location of the output files created by this class. If not given, it will be
+            taken from the config.yml file.
+        configData: dict.
+            Dict containing the configurable parameters.
+        configFile: str or Path
+            Path of the yaml file containing the configurable parameters.
+        singleMirrorMode: bool
+            True for single mirror simulations.
+        '''
+        self._logger = logging.getLogger(__name__)
+        self._logger.debug('Init SimtelRunnerRayTracing')
+
+        super().__init__(
+            label=label,
+            simtelSourcePath=simtelSourcePath,
+            filesLocation=filesLocation
+        )
+
+        self.telescopeModel = self._validateTelescopeModel(telescopeModel)
+        self.label = label if label is not None else self.telescopeModel.label
+
+        # File location
+        self._baseDirectory = io.getOutputDirectory(
+            self._filesLocation,
+            self.label,
+            'ray-tracing'
+        )
+        self._baseDirectory.mkdir(parents=True, exist_ok=True)
+
+        self._singleMirrorMode = singleMirrorMode
+
+        # RayTracing - default parameters
+        self._repNumber = 0
+        self.RUNS_PER_SET = 1 if self._singleMirrorMode else 20
+        self.PHOTONS_PER_RUN = 10000
+
+        # Loading configData
+        _configDataIn = gen.collectDataFromYamlOrDict(configFile, configData)
+        _parameterFile = io.getDataFile('parameters', 'simtel-runner-ray-tracing_parameters.yml')
+        _parameters = gen.collectDataFromYamlOrDict(_parameterFile, None)
+        self.config = gen.validateConfigData(_configDataIn, _parameters)
+
+        self._loadRequiredFiles()
+
+    def _loadRequiredFiles(self):
+        '''
+        Which file are required for running depends on the mode.
+        Here we define and write some information into these files. Log files are always requires.
+        '''
+
+        self._corsikaFile = self._simtelSourcePath.joinpath('run9991.corsika.gz')
```

Comment:
Not sure I get this: file name is hard wired here?

Reply:
Yes, this is a empty CORSIKA file that used to come in the sim_telarray package. I realized recently that it is actually not there for the recent versions.
The file not existing is not a problem for the ray tracing simulations itself. The only problem is that the sim_telarray execution returns an error code, so we can't actually identify real execution errors.

I created an issue to go back to this later.